### PR TITLE
#159987411 User can tag an article and view articles

### PR DESCRIPTION
**Pull request by jojitoon**
#### What does this PR do?
Implement functionality to allow user view tags and associated articles

#### Description of Task to be completed?

- modify database relationship
- implement the functionality to create tags from article creation
- implement the functionality to view all articles
- implement the functionality to view a tag with all associated articles

#### How should this be manually tested?

- Clone the repository.
- Checkout the `ft-article-tagging-159987411`  branch.
- Run npm install on your local machine.
- Run the db migrations using npm run migrate.
- Start the server using npm run start_dev.
- Using postman or any similar. get  a token upon registration and use it to test these routes:
  `localhost:8000/api/v1/tags`
`localhost:8000/api/v1/tags/:tagName`

#### Any background context you want to provide?
This feature will enable a user to view all tags and articles associated with the tags.

#### What are the relevant pivotal tracker stories?
#159987411

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -40,11 +41,13 @@ app.use('/api/v1/users', userRoutes);
 app.use('/api/v1/users/auth', authRoutes);
 app.use('/api/v1/articles', articleRoutes);
 app.use('/api/v1/cases', caseRoutes);
+app.use('/api/v1/tags', tagRoute);
 
 app.get('/', (req, res) => res.status(200).jsend.success({
   message: 'Welcome to the sims program'
 }));
 
+
```
Comment (iAmao) [inline server/app.js]: Unnecessary blank line

File: server/controllers/articleController.js [hunk 2]
```diff
@@ -43,10 +45,18 @@ const articlesController = {
       description: fields.description,
       body: fields.body,
       imageUrl
-    }).then(createdArticle => res.status(201).jsend.success({
-      article: createdArticle,
-      message: 'Article published successfully'
-    })).catch(err => res.status(500).jsend.fail({
+    }).then((createdArticle) => {
+      // checks if tags exist
+      const tags = !fields.tags || fields.tags === '' ? []
```
Comment (iAmao) [inline server/controllers/articleController.js]: When working with ternary operators try modeling it like this:
```
condition
 ? action
 : action
```

File: server/controllers/tagController.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import models from '../models';
+
+const { Articles, Tags } = models;
+
+
```
Comment (iAmao) [inline server/controllers/tagController.js]: Unnecessary blank lines

File: server/controllers/tagController.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+const tagController = {
+
```
Comment (iAmao) [inline server/controllers/tagController.js]: Unnecessary blank lines

File: server/migrations/20180911162010-create-tagging.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+
+
```
Comment (iAmao) [inline server/migrations/20180911162010-create-tagging.js]: Unnecessary blank lines

File: server/models/tagging.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+
+
+module.exports = (sequelize, DataTypes) => {
+  const Taggings = sequelize.define('Taggings', {
```
Comment (iAmao) [inline server/models/tagging.js]: Follow naming convention of pivot Table here

**Review by iAmao — CHANGES_REQUESTED** [review]
This is good nice job

File: server/models/articlesTags.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+module.exports = (sequelize, DataTypes) => {
+  const Taggings = sequelize.define('ArticlesTags', {
```
Comment (iAmao) [inline server/models/articlesTags.js]: this is still called `Tagging`

**Review by iAmao — DISMISSED** [review]

**Review by tomiadebanjo — APPROVED** [review]

**Pull request merged**